mars/worker/__main__.py

```diff
@@ -59,15 +59,19 @@ def create_pool(self, *args, **kwargs):
 
         options.worker.cpu_process_count = int(self.args.cpu_procs or resource.cpu_count())
         options.worker.io_process_count = int(self.args.io_procs or '1')
+
+        if self.args.phy_mem:
+            mem_total = self._calc_size_limit(self.args.phy_mem, mem_stats.total)
+        else:
+            mem_total = mem_stats.total
+
         options.worker.physical_memory_limit_hard = self._calc_size_limit(
-            self.args.phy_mem or options.worker.physical_memory_limit_hard, mem_stats.total
+            options.worker.physical_memory_limit_hard, mem_total
```

Comment:
those memory calculations may be move to `WorkerService`

Reply:
> those memory calculations may be moved to `WorkerService`

Currently local cluster use 20% of shared memory.

Reply:
This calculation is in main instead of `WorkerService.start` before.

Reply:
Already moved and unified.

Reply:
Currently options should be set before actor pool starts. Thus these configurations CANNOT be moved inside start_service. Further refactoring are welcomed.